test/libsolidity/smtCheckerTests/operators/index_access_side_effect.sol

```diff
@@ -0,0 +1,16 @@
+pragma experimental SMTChecker;
+
+contract C {
+	uint[] a;
+	function h() internal returns (uint[] storage) {
+		if (a[2] == 0)
```

Comment:
Can you add the same test with `if (a[2] == 4)` here, so we can see assertion is safe

Reply:
I could add that test, but it still reports assertion violation because the smtchecker doesn't support aliasing properly. That is, reference types are actually passed by value in the smtchecker, and it doesn't keep track of pointers/references. So in that case, it doesn't know that the return from `h()` is actually the same as the state variable `a`.
To still be sound, we erase knowledge about reference types when one is assigned, so the assertion violation is reported as a false positive.

Reply:
Thanks for explanation :-)